secure_message/resources/threads.py

```diff
@@ -113,16 +116,26 @@ def get():
             messages = add_users_and_business_details(messages)
         return jsonify({"messages": messages, "_links": links})
 
+    @staticmethod
+    def _validate_request(request_args, user):
+        if request_args.my_conversations and user.is_respondent:
+            logger.error('My conversations option not available to respondents')
+            raise BadRequest(description="My conversations option not available to respondents")
+
 
 class ThreadCounter(Resource):
     """Get count of all open or closed messages"""
     @staticmethod
     def get():
         surveys = request.args.getlist('survey')
         is_closed = bool(strtobool(request.args.get('is_closed', 'False')))
+        my_conversations = bool(strtobool(request.args.get('my_conversations', 'False')))
+        cc = request.args.get('cc', None)
+        ce = request.args.get('ce', None)
+        ru = request.args.get('ru_id', None)
```

Comment:
what do we do with this RU? I mean, do we have protection to stop people from guessing other RUs and seeing stuff they shouldn't?

Reply:
Only an internal user hits this end point . On a respondent threads lookup the respondent uuid is a param in the search . They are free to guess an ru_id , its a uuid, but if they do then they only see their conversations.